cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java

```diff
@@ -140,4 +146,130 @@ private RuntimeArguments() {
     return result;
   }
 
+  private static final Pattern DATE_PATTERN = Pattern.compile("\\$\\{([^\\{\\}]+)\\}");
+
+  private static final Pattern OFFSET_PATTERN = Pattern.compile(":\\$\\{(.*)\\}$");
+
+  /**
+   * Resolves the {@link Schedule} arguments based on the rules specified in the argument values and
+   * logical start time of the {@link Schedule}. Rules specified in the property values must be enclosed
+   * by "${" and "}" and should conform to the patterns specified by {@link SimpleDateFormat}.
+   *
+   * e.g. Consider following properties are specified for the {@link Schedule}
+   * <p>
+   *   input.file=purchase_data_${MMddyyyy}_${Ha}.input
+   *   output.file=purchase_data_${dEyyyy}.output
+   * </p>
+   *
+   * When {@link Schedule} triggers at logical start time say '1422957600' (February 3, 2015 at 2:00:00 AM),
```

Comment:
If the {@link Schedule} triggers at a logical start time of '1422957600' (February 3, 2015 at 2:00:00 AM),
the rules in the {@link Schedule} arguments will be evaluated and these properties will be passed to
the triggered program: